pkg/scheduler/scheduler.go

```diff
@@ -647,6 +672,13 @@ func (s *Scheduler) requeueAndUpdate(ctx context.Context, e entry) {
 		// Failed after nomination is the only reason why a workload would be requeued downstream.
 		e.requeueReason = queue.RequeueReasonFailedAfterNomination
 	}
+
+	if s.queues.QueueSecondPassIfNeeded(ctx, e.Obj) {
+		log.V(2).Info("Workload re-queued for second pass", "workload", klog.KObj(e.Obj), "clusterQueue", klog.KRef("", string(e.ClusterQueue)), "queue", klog.KRef(e.Obj.Namespace, string(e.Obj.Spec.QueueName)), "requeueReason", e.requeueReason, "status", e.status)
+		s.recorder.Eventf(e.Obj, corev1.EventTypeWarning, "SecondPassFailed", api.TruncateEventMessage(e.inadmissibleMsg))
+		return
+	}
+
 	added := s.queues.RequeueWorkload(ctx, &e.Info, e.requeueReason)
```

Comment:
Do we want to add Workloads that require second pass of the scheduler to the regular CQ heap, if the second pass didn't succeed right away?

Reply:
I think it is better not too add them to the heap, because it is a lot of refactoring to return multiple workloads from heap, and I wouldn't want the "second pass" workloads to starve the regular "tip" of the heap. So, I use secondPassQueue to keep them separaterly.